subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/UnexportMainSymbol.java

```diff
@@ -71,7 +71,8 @@ public File getMainObject() {
     public void unexport() {
         final File mainObjectFile = getMainObject();
         if (mainObjectFile != null) {
-            final File relocatedMainObject = outputFile.file(mainObjectFile.getName()).get().getAsFile();
+            final File relocatedMainObject = outputDirectory.file(mainObjectFile.getName()).get().getAsFile();
+            relocatedMainObject.getParentFile().mkdirs();
```

Comment:
Could you have a getter for the directory and annotate it with @OutputDirectory so you don't have to mkdirs here? Does that even make sense? Probably what you have there make senses.

Reply:
Yeah, there are two things going on:

- To link this object optionally (when there are no sources), we need a `FileTree`
- To get the directory created automatically, we need a `@OutputDirectory`.

We could do something like:
```
@Internal FileCollection getObjects()
@OutputDirectory Provider<Directory> getOutputDirectory()
```

I can try that.